elements/lisk-chain/src/data_access/transaction_interface_adapter.ts

```diff
@@ -24,25 +26,52 @@ export class TransactionInterfaceAdapter {
 	public constructor(registeredTransactions: RegisteredTransactions = {}) {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 		this._transactionClassMap = new Map();
+		codec.addSchema(BaseTransaction.BASE_SCHEMA);
 		Object.keys(registeredTransactions).forEach(transactionType => {
+			const transaction = registeredTransactions[transactionType];
 			this._transactionClassMap.set(
 				Number(transactionType),
-				registeredTransactions[transactionType],
+				transaction,
 			);
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+			const { ASSET_SCHEMA } = this._transactionClassMap.get(Number(transactionType));
+			codec.addSchema(ASSET_SCHEMA);
 		});
 	}
 
-	public fromJSON(rawTx: TransactionJSON): BaseTransaction {
+	// First encode message asset and then encode base message
+	public encode(message: BaseTransaction): Buffer {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-		const TransactionClass = this._transactionClassMap.get(rawTx.type);
+		const TransactionClass = this._transactionClassMap.get(message.type);
```

Comment:
I think it's good to have this function, but for encode, isnt is just calling `getBytes` of the message?

Reply:
👍